File: 1829-maximum-xor-for-each-query/1829-maximum-xor-for-each-query.cpp
class Solution {
public:
    vector<int> getMaximumXor(vector<int>& nums, int maximumBit) {
        int n = nums.size();
        int xo = 0;
        vector<int> res (n, 0);
        for (int x: nums) {
            xo ^= x;
        }
        int me = (1 << maximumBit) - 1;
        for (int i = n-1; i >= 0; i --) {
            res[n-i-1] = (me ^ xo);
            xo ^= nums[i];
        }
        return res;
    }
};

File: 1829-maximum-xor-for-each-query/1829-maximum-xor-for-each-query.py
class Solution:
    def getMaximumXor(self, nums: List[int], maximumBit: int) -> List[int]:
        res = []
        xor = 0
        for i in nums:
            xor ^= i
        n = len(nums)
        k = (1 << maximumBit) - 1
        for i in range(n-1,-1,-1):
            res.append(k^xor)
            xor ^= nums[i]
        return res